subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdaterTest.groovy

```diff
@@ -72,12 +72,19 @@ abstract class AbstractFileWatcherUpdaterTest extends Specification {
         }
     }
 
+    List<File> movedPaths = []
+    AbstractFileWatcherUpdater.MovedWatchedDirectoriesSupplier movedWatchedDirectoriesSupplier = { -> movedPaths }
+
     def setup() {
         updater = createUpdater(watcher, watchableHierarchies)
     }
 
     abstract int getIfNonHierarchical()
 
+    int getIfHierarchical() {
+        return 1 - getIfNonHierarchical()
+    }
```

Comment:
I think these two methods should have more context. Either Javadoc or better names.

Reply:
I added Javadocs